src/main/java/com/arpnetworking/metrics/mad/Aggregator.java

```diff
@@ -129,8 +146,12 @@ public void notify(final Observable observable, final Object event) {
             return;
         }
 
+        // TODO(ville): Should the record contain a key instead of just annotations?
+        // ^ This raises the bigger question of metric name as part of the key.
+        // (at the moment it's not taking advantage of same key-space across metrics in bucket)
         final Record record = (Record) event;
-        _actor.tell(record, ActorRef.noSender());
+        final int actorIndex = (record.getDimensions().hashCode() & 0x7FFFFFFF) % _actors.size();
```

Comment:
hashCode isn't particularly uniform for partitioning purposes. Perhaps MD5 here instead?

More of a nit, the modulo will bias messages to the actors at the front of the set. That's probably a very small issue here given the (relatively) small number of actors.

Reply:
Yes, I totally agree.

The flip side is that computing the MD5 hash of the dimensions will result in additional overhead.

The skew from the modulo is pretty minimal as a percent of the keyspace. 

I will add a comment regarding these issues.